### PR TITLE
Update sonobuoy version

### PR DESCRIPTION
**Pull request by pablochacin**
## Why is this PR needed?

Conformance tests are failing with error: `The maximum supported Kubernetes version is 1.17.99, but the server version is v1.18.2. Sonobuoy will continue but unexpected results may occur.`

Fixes https://github.com/SUSE/avant-garde/issues/1708

## What does this PR do?

Pass sonobuoy versio as a paramter to conformace tests.

Update sonobuoy version for master branch to one that supports k8s 1.18

Removes unnecessary checkout logic in the pipeline that prevents testing against the job agains a pr branch. 

# Note to reviewers

The PR was tested [running the conformance test job against the pr branch](https://ci.suse.de/job/caasp-jobs/job/conformance/job/v5/job/vmware/22/)

Screenshot from 2020-06-16 11-41-49

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

File: ci/jenkins/templates/conformance-template.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@
     days-to-keep: 30
     branch: master
     schedule: '3-5'
+    sonobuoy_version: 'v0.18.3'
```
Comment (dirkmueller) [inline ci/jenkins/templates/conformance-template.yaml:8]: without https://build.suse.de/request/show/220338 being accepted this needs to read 0.18.2

Comment (pablochacin) [reply ci/jenkins/templates/conformance-template.yaml]: Good catch. Fixed.

**Comment by jordimassaguerpla** [comment]
Do we have to update the sonobuoy package in Devel:CaaSP:5 ?

**Comment by dirkmueller** [comment]
no, 1.18.2 is already in Devel:CaaSP:5. ideally somebody would review ibs rq 220338 though and then this could refer to the latest version instead.

**Pull request merged**